### PR TITLE
MAGECLOUD-4102: Refactor existing code to fit the new scenario-based deployments

### PR DESCRIPTION
**Pull request by BaDos**
### Fixed Issues (if relevant)

https://magento2.atlassian.net/browse/MAGECLOUD-4102

### Manual testing scenarios

1. ...
2. ...

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

File: src/Step/ProcessException.php [hunk 1]
```diff
@@ -3,7 +3,7 @@
  * Copyright © Magento, Inc. All rights reserved.
```
Comment (bbatsche) [inline src/Step/ProcessException.php]: Shouldn't this be `StepException` now?

**Review by bbatsche — CHANGES_REQUESTED** [review]
One question about `ProcessException` but otherwise looks good

**Comment by oshmyheliuk** [comment]
@BaDos please merge `develop` branch

File: src/Step/ProcessException.php (hunk 1 above)
Comment (BaDos) [reply src/Step/ProcessException.php]: Renamed

File: src/App/Container.php [hunk 2]
```diff
@@ -119,13 +119,13 @@ public function __construct(string $toolsBasePath, string $magentoBasePath)
             \Magento\MagentoCloud\Config\Database\MergedConfig::class
         );
 
-        $this->container->when(DeployProcess\InstallUpdate\ConfigUpdate\Urls::class)
-            ->needs(ProcessInterface::class)
+        $this->container->when(DeployStep\InstallUpdate\ConfigUpdate\Urls::class)
```
Comment (oshmyheliuk) [inline src/App/Container.php:122]: Why this configuration still in the Contaner file? Looks like it should be configured in scenario XML file.

Comment (oshmyheliuk) [reply src/App/Container.php:122]: @shiftedreality Is it possible configure steps for sub-steps in senario.xml files?

Comment (shiftedreality) [reply src/App/Container.php:122]: It's currently too complicated for XML file, we'll have to refactor this case later

**Review by oshmyheliuk — APPROVED** [review]

**Review by bbatsche — APPROVED** [review]

**Comment by shiftedreality** [comment]
No QA needed

**Pull request merged**